contracts/voting/dao-voting-native-staked/src/contract.rs

```diff
@@ -165,16 +292,19 @@ pub fn execute_unstake(
                 .map_err(|_e| ContractError::InvalidUnstakeAmount {})
         },
     )?;
+    let hook_msgs = unstake_hook_msgs(deps.storage, info.sender.clone(), amount)?;
 
     let config = CONFIG.load(deps.storage)?;
+    let denom = DENOM.load(deps.storage)?;
     match config.unstaking_duration {
         None => {
             let msg = CosmosMsg::Bank(BankMsg::Send {
                 to_address: info.sender.to_string(),
-                amount: coins(amount.u128(), config.denom),
+                amount: coins(amount.u128(), denom),
             });
             Ok(Response::new()
                 .add_message(msg)
+                .add_submessages(hook_msgs)
                 .add_attribute("action", "unstake")
                 .add_attribute("from", info.sender)
                 .add_attribute("amount", amount)
```

Comment:
I know this is not a part of the pr but what happens if in `Some(duration)` case we exceed the max claims?

Reply:
Worth an investigation.